phoenix-core/src/main/java/org/apache/phoenix/schema/export/SchemaRegistryRepositoryFactory.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.phoenix.schema.export;
+
+import org.apache.hadoop.conf.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+public final class SchemaRegistryRepositoryFactory {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SchemaRegistryRepository.class);
+    private static SchemaRegistryRepository exporter;
+
+    public synchronized static SchemaRegistryRepository getSchemaRegistryRepository(Configuration conf)
+            throws IOException {
+        if (exporter != null) {
+            return exporter;
+        }
+        try {
+            String className = conf.get(SchemaRegistryRepository.SCHEMA_REGISTRY_IMPL_KEY);
+            if (className == null) {
+                exporter = new DefaultSchemaRegistryRepository();
```

Comment:
How about moving this in try ( so that it is auto closed?

Reply:
The factory can't close it because the exporter hasn't been used yet. But good point that someone has to close it. Will add a method to close the factory and have the coproc call it during stop()

Reply:
Added the factory close method as I described